### ISSUE TITLE
Face clustering: 42 faces left

### ISSUE THREAD

**Issue by 31337-4554551n** [nextcloud/recognize#1200]
Seeing an issue where 42 faces are never being processed.

Unsure if separate or related issue, as new photos are taken, most are being recognised, but in a few cases (maybe 1/5) the face isn't being detected. This is selfies/photos with obvious people.

Original discussion:
```
Face recognition: 0 Queued files, Last classification: 12 hours ago, Scheduled background jobs: 0, 
```

```
 Face clustering: 42 faces left to cluster, Last clustering run: 12 hours ago, Scheduled background jobs: 0,
A minimum of 120 faces per user is necessary for clustering to kick in
```
If I'm reading this correctly, I have 78 pics with faces detected, but they won't be added into their correct people until I take another 42 photos with faces in them?

Is there any way to shrink those batches?
I definitely know I have photos with peoples faces that aren't showing up under people... and idk how to force em

[edit] to clarify I am using recognize to feed into memories

_Originally posted by @31337-4554551n in https://github.com/nextcloud/recognize/discussions/1167_

**Comment by marcelklehr** [comment]
> but in a few cases (maybe 1/5) the face isn't being detected

This could be due to using WASM mode, or because of shortcomings of the face detection model. It's not easily fixable.

Unless the missing faces are in the 42 faces that are left.

How many pictures do you have overall?

**Comment by 31337-4554551n** [comment]
I am using a WASM module, this is on the linuxserver.io docker, it's alpine based, and I spent a long time trying to get it to work properly, but it's just missing too many dependencies for GNU lib C, there's a github with it compiled, but then it complains about missing nodeJS and other stuff.

A little under 10,000 photos on one account, and a little under 9000 on another.

**Comment by marcelklehr** [comment]
> A little under 10,000 photos on one account, and a little under 9000 on another.

This is peculiar. With so many photos there should be enough detected faces to sample in enough to get over the 120 faces limits. Except... how many people do you have? If you have all detections grouped into very few people this might explain why it's not overcoming the sample limit.

**Comment by 31337-4554551n** [comment]
It's not detecting the 42. And it's not detecting existing faces, like I'll go out, take 3 photos of the gf,and it'll import 2 for example.
There's be maybe 20-30 faces. There were more, but I deleted them when it picks out like random people in a crowd or something.

**Comment by marcelklehr** [comment]
20-30 faces or 20-30 face clusters (people)?

**Comment by 31337-4554551n** [comment]
20-30 clusters.
1000-2000 pictures of faces.

**Comment by 31337-4554551n** [comment]
What more information is being waited on? I gave the last response

20-30 clusters.
1000-2000 pictures of faces.

**Comment by marcelklehr** [comment]
Yes, I'm sorry, I didn't remove the "waiting for more information" tag

**Comment by marcelklehr** [comment]
I was able to reproduce this and fix it. The fix will be in the next release
